pkg/jsonmessage/jsonmessage.go

```diff
@@ -189,13 +189,9 @@ func DisplayJSONMessagesStream(in io.Reader, out io.Writer, terminalFd uintptr,
 				if isTerminal {
 					fmt.Fprintf(out, "\n")
 				}
-			} else {
-				diff = len(ids) - line
 			}
-			if isTerminal {
-				// NOTE: this appears to be necessary even if
-				// diff == 0.
-				// <ESC>[{diff}A = move cursor up diff rows
+			diff = len(ids) - line
```

Comment:
I think I have misunderstood what you are suggesting because if `ok` is `false` then `line` should be `1` not `0` (that's the main thrust of this PR) and the calculation of `diff` is independent of the value of `ok` (after this PR).